### PR TITLE
GTEST/UCP: Introduce CM_USE_ALL_DEVICES=y variant in sockaddr tests

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Introduce `CM_USE_ALL_DEVICES=y` variant in sockaddr tests.

## Why ?

For better test coverage.

## How ?

1. Add new test modifier `TEST_MODIFIER_CM_USE_ALL_DEVICES` in the same way as we have for `TEST_MODIFIER_MT`.
2. if `TEST_MODIFIER_CM_USE_ALL_DEVICES` set, set `CM_USE_ALL_DEVICES` to `"y"`, otherwise to `"no"`
3. Skip `connect_and_fail_wireup` and `compare_cm_and_wireup_configs` test if  `CM_USE_ALL_DEVICES` is `"no"`

### CODE REVIEW

**Comment by dmitrygx** [comment]
@evgeny-leksikov @hoopoepg could you review pls?

**Review by hoopoepg — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls?

**Review by evgeny-leksikov — COMMENTED** [review]
what's the reason to set USE_ALL_DEVICES=y if it's enabled by default? AFAIU we have missed coverage for USE_ALL_DEVICES=n

**Comment by dmitrygx** [comment]
> what's the reason to set USE_ALL_DEVICES=y if it's enabled by default?

it is needed to make sure `UCX_USE_ALL_DEVICES=y` is set for this test in case of a user runs gtests with `UCX_CM_USE_ALL_DEVICES=n`.

> AFAIU we have missed coverage for USE_ALL_DEVICES=n

yes, it is

**Comment by evgeny-leksikov** [comment]
> it is needed to make sure `UCX_USE_ALL_DEVICES=y` is set for this test in case of a user runs gtests with `UCX_CM_USE_ALL_DEVICES=n`

in this case I think it would be better to exclude these tests from user perspective

**Comment by dmitrygx** [comment]
> > it is needed to make sure `UCX_USE_ALL_DEVICES=y` is set for this test in case of a user runs gtests with `UCX_CM_USE_ALL_DEVICES=n`
> 
> in this case I think it would be better to exclude these tests from user perspective

what does it mean to exclude?
this is common practise: we have other tests where we don't assume to use RNDV, so they set `RNDV_THRESH=inf`.

**Comment by evgeny-leksikov** [comment]
> > > it is needed to make sure `UCX_USE_ALL_DEVICES=y` is set for this test in case of a user runs gtests with `UCX_CM_USE_ALL_DEVICES=n`
> > 
> > 
> > in this case I think it would be better to exclude these tests from user perspective
> 
> what does it mean to exclude?
> this is common way: we have other tests where we don't assume to use RNDV, so they set `RNDV_THRESH=inf`.

in case of RNDV we change default auto selection behavior to make the test behavior the same on all platforms, `UCX_USE_ALL_DEVICES` logic is constant.

> what does it mean to exclude?
> this is common way: we have other tests where we don't assume to use RNDV, so they set `RNDV_THRESH=inf`.

If I set some var outside of gtest, I expect that it can affect some tests,  so I exclude them by gtest_filter 

IMO we should add a parameter for testing `UCX_USE_ALL_DEVICES=y/n` and skip these tests if `UCX_USE_ALL_DEVICES=n`

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 1]
```diff
@@ -916,6 +931,10 @@ UCS_TEST_P(test_ucp_sockaddr, compare_cm_and_wireup_configs) {
     ucp_worker_cfg_index_t cm_ep_cfg_index, wireup_ep_cfg_index;
     ucp_ep_config_key_t *cm_ep_cfg_key, *wireup_ep_cfg_key;
 
+    if (!cm_use_all_devices()) {
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc:934]: common helper?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_sockaddr.cc:934]: better to move to `UCS_TEST_SKIP_COND_P`

Comment (dmitrygx) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: good catch, done

**Comment by dmitrygx** [comment]
@yosefe @evgeny-leksikov could you review pls?

**Review by evgeny-leksikov — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe ok to merge?

**Pull request merged**